File: Tool/PreUtil/src/output/javascript/medium/MaximumSubarray.js
/**https://leetcode.com/problems/maximum-subarray/ */
//Given an integer array nums, find the subarray with the largest sum, and return its sum.


/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let max = nums[0];
    let sum = nums[0];
    for(let i = 1; i < nums.length; i++){
        sum = Math.max(nums[i], sum + nums[i]);
        max = Math.max(max, sum);
    }
    return max;
    
};

    

File: Tool/PreUtil/src/output/c/medium/MaximumSubarray.c
/**https://leetcode.com/problems/maximum-subarray/ */
//Given an integer array nums, find the subarray with the largest sum, and return its sum.


int maxSubArray(int* nums, int numsSize){
    int max = nums[0];
    int sum = 0;
    for(int i = 0; i < numsSize; i++){
        sum += nums[i];
        if(sum > max){
            max = sum;
        }
        if(sum < 0){
            sum = 0;
        }
    }
    return max;

}

    